src/document.rs

```diff
@@ -510,11 +510,11 @@ impl Document {
         }
     }
 
-    /// Attempts to encode the `Document` into a byte stream.
-    pub fn encode<W: Write + ?Sized>(&self, writer: &mut W) -> EncoderResult<()> {
+    /// Attempts to serialize the `Document` into a byte stream.
+    pub fn serialize_doc<W: Write + ?Sized>(&self, writer: &mut W) -> crate::ser::Result<()> {
```

Comment:
Since `Document` implements the `Serialize` trait [here](https://github.com/mongodb/bson-rust/blob/ed1740acde6b1f21007af8872d275404e6f415e1/src/encoder/serde.rs#L45), and calls `serialize` [here](https://github.com/mongodb/bson-rust/blob/ed1740acde6b1f21007af8872d275404e6f415e1/src/encoder/serde.rs#L69), the compiler wasn't happy with this method being named `serialize`. (I had no idea Rust doesn't allow method overloading, interesting.)

Reply:
Good catch, I hadn't thought of this conflict!. Technically you _could_ have two different methods with the same name if one of them is from a trait, but it's not traditional overloading because there's no disambiguation at the call site based on the parameters; you'd only have access to one of methods for a given variable based on what type it is. Here's an example of that: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=996a1b2bab9c7a8602de0098de00012c

I think I might prefer the name `serialize_to` here; when calling it, I think it would read well, e.g. `doc.serialize_to(&mut writer)?`.

Reply:
I agree, less redundant than having `doc` in the name

Reply:
I like `serialize_to`, but we might also want to consider something like `to_writer` and `from_reader` like in [`serde_json`](https://docs.rs/serde_json/1.0.53/serde_json/fn.to_writer.html) and [`serde-xml-rs`](https://docs.rs/serde-xml-rs/0.4.0/serde_xml_rs/ser/fn.to_writer.html).

Reply:
I'm fine with `from_reader` and `to_writer`.

Reply:
I'm fine with both, I'll leave as-is with `serialize_to` unless anyone feels super strongly either way.

Reply:
You can change it to Patrick's suggestions; sorry if I was unclear

Reply:
oops sorry I didn't see that first comment! I had this window sitting open for a while and didn't realize you need to refresh to see new comments. will do

Reply:
No worries; github does require a lot of refreshes that it should probably trigger on its own